clang/test/Driver/clang-offload-wrapper-exe-preview.cpp

```diff
@@ -0,0 +1,211 @@
+// End-to-end clang-offload-wrapper executable test: check that -batch options
+// works, and that the tool generates data properly accessible at runtime.
+
+// Prepare test data. Create the first binary image.
+// RUN: echo -e -n 'device binary image1\n' > %t.bin
+// RUN: echo -e -n '[Category1]\nint_prop1=1|10\n[Category2]\nint_prop2=1|20\n' > %t.props
+// RUN: echo -e -n 'kernel1\nkernel2\n' > %t.sym
+
+// Create the second binary image with byte array property values.
+// RUN: echo -e -n 'device binary image2\n' > %t_1.bin
+// RUN: echo -e -n '[Category3]\n' > %t_1.props
+// RUN: echo -e -n 'kernel1=2|IAAAAAAAAAQA\n' >> %t_1.props
+// RUN: echo -e -n 'kernel2=2|oAAAAAAAAAw///3/wB\n' >> %t_1.props
+
+// Create the batch file input for the wrapper.
+// RUN: echo '[Code|Properties|Symbols]' > %t.batch
+// RUN: echo %t.bin"|"%t.props"|"%t.sym >> %t.batch
+// RUN: echo %t_1.bin"|"%t_1.props"|" >> %t.batch
+
+// Generate "gold" output.
+// RUN: cat %t.bin %t.props %t.sym > %t.all
+
+// Create the wrapper object.
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+/// TODO: Remove -fpreview-breaking-changes from the command line, when removing the macro.
+// RUN: clang-offload-wrapper -kind=sycl -target=TARGET -format=native -batch %t.batch -o %t.wrapped.bc -fpreview-breaking-changes
+#endif // __INTEL_PREVIEW_BREAKING_CHANGES
+// RUN: llc --filetype=obj %t.wrapped.bc -o %t.wrapped.o
+
+// Compile & link the test with the wrapper.
+// RUN: %clangxx %t.wrapped.o %s -o %t.batch.exe -v
+
+// Run and check ignoring white spaces.
+// RUN: %t.batch.exe > %t.batch.exe.out
+// RUN: diff -b %t.batch.exe.out %t.all
+
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+#include <string>
+
+// Data types created by the offload wrapper and inserted in the wrapper object.
+// Match those defined in SYCL runtime Plugin Interface.
+struct _pi_offload_entry_struct {
```

Comment:
Can you elaborate what this means?

Reply:
I will update description. `pi` is legacy term. I will update the whole test to not use it. All data structures will match to SYCL Runtime.